### ISSUE TITLE
Add pre-compiled mlpack download alternative

### ISSUE THREAD

**Issue by gmanlan** [mlpack/mlpack.org#7]
The new website looks fantastic!
I noticed that we only offer a single download option (just the src) which maybe adds little value to new users arriving to mlpack (especially for users without the time to compile mlpack and all the dependencies for a 'try out'). Since the builds are already producing ZIP/MSI artifacts, it would be nice to add one stable version of a pre-compiled mlpack as an alternative download option at http://mlpack.org/

**Comment by rcurtin** [comment]
Thanks!  You're right, I definitely agree, maybe adding a Windows zip to download would be a nice addition.  We can just pull the artifacts off the AppVeyor build.  To keep the number of links short, I'd like to pick only one to add... would you say that the MSI would be best, or would `mlpack-windows.zip` be better?

It might be good to also add some simple "other installation options" link or something that can point users at package repositories or similar; I'll think about how I can add that too.

**Comment by gmanlan** [comment]
The MSI is at least for Windows, a friendlier way of "installing" new software. When we created the MSI installer it was mainly because we already had a super easy way of installing for Linux but not for Windows. Therefore, the MSI would be more familiar to Windows users. On the other hand, the zip might be safer considering that we don't have a certificate to sign the installer (which may scare some users).

Here is my personal experience:
Every time I want to quickly try a new library, I tend to avoid (or keep at the bottom of my list) libraries that oblige me to compile. This is mainly because I don't have much time and if I'm experimenting, I would go with the fastest option first. Because pre-compiled binaries are very rare, sometimes I get these from somebody else (a.k.a. a co-worker) that already went through the compilation nightmare. Only after I have tested the library and I like it, I will go and spend the extra effort to properly build/adopt it.
In conclusion, the easier and friendlier it is, the better chances of people adopting the library.

When you compare it with other ML libs (imagine how easy is to pip install something nowadays, or how easy is to gradle/jar-download SMILE binaries), the adoption user experience is not the best for us (also, think about how many emails we receive in the email list asking about compilation stuff).

We can stick with the windows zip, but I would strongly suggest providing a path for discovery of the MSI. As far as I know, nobody knows that there is an MSI installer for mlpack (you have to dig into the build server and retrieve a temporal artifact...).

Let me know if it makes sense.

**Comment by rcurtin** [comment]
This is great commentary and I really appreciate it.  It's easy to forget what the experience is like for a new user. :)

I'll add the MSI, and try to think about how I can automate that part of the release process.  (It would be nice to also provide generic Linux binaries and OS X binaries too.)  I think it would also be easy to add a link to another page discussing how to get it through package managers for various distributions.

**Comment by gmanlan** [comment]
Sounds good. We can also explore options such as Chocolatey, but that's an story for a different day.
Let me know if you need help with this. I'm not familiar with the website code but I can try if you want.

**Comment by rcurtin** [comment]
Oops, I let this sit too long, but anyway I did actually update the website and it's listing the MSI now, so I think we can resolve this.  I don't think just the MSI is a permanent solution (maybe some instructions might be nice to go with it) but if you are refactoring the website and making it modern and clean and pretty, then I think we can wait until then. :)
